### PR TITLE
Added an example how to use the new splitter resize event

### PR DESCRIPTION
**Pull request by skeetor**
I added now an example on how to use the resize event.

I'm terrible sorry, but when I added the example, I noticed that I missed some renaming before. Originally I wanted the event to use UPDATE in the name instead of RESIZE, but when I thought about it's meaning I figured that RESIZE is better. I changed all occurrences, but I missed one file which didn't show up in my own code and only broke the build when I compiled the sample.

I fixed it now along with the sample.

### CODE REVIEW

File: samples/splitter/splitter.cpp [hunk 1]
```diff
@@ -116,11 +118,18 @@ class MyFrame: public wxFrame
...
+    void SetSashPos(int pos) { m_sashPos = pos; }
+    int GetSashPos() const { return m_sashPos; }
+
 private:
     wxWindow *m_left, *m_right;
 
     wxSplitterWindow* m_splitter;
     wxWindow *m_replacewindow;
+    int m_sashPos;
+    bool m_lockSash:1;
```
Comment (vadz) [inline samples/splitter/splitter.cpp:132]: Minor, but there is really no need to optimize this here, so
```suggestion
    bool m_lockSash;
```

File: samples/splitter/splitter.cpp [hunk 2]
```diff
@@ -263,6 +277,9 @@ MyFrame::MyFrame()
     splitMenu->Append(SPLIT_SETGRAVITY,
                       "Set &gravity\tCtrl-G",
                       "Set gravity of sash");
+    splitMenu->Append(SPLIT_LOCKSASH,
```
Comment (vadz) [inline samples/splitter/splitter.cpp:280]: Shouldn't this be a check item, i.e.
```suggestion
    splitMenu->AppendCheckItem(SPLIT_LOCKSASH,
```

File: samples/splitter/splitter.cpp [hunk 3]
```diff
@@ -521,6 +545,23 @@ void MySplitterWindow::OnPositionChanged(wxSplitterEvent& event)
...
+
+void MySplitterWindow::OnPositionResize(wxSplitterEvent &event)
+{
+    // When the splitter is resizing we only allow the sash to be moved
+    // if it is not locked. Otherwise we hold it at the position
+    // the user specified by manually dragging.
+    if (m_frame->IsSashLocked())
+        event.SetSashPosition(m_frame->GetSashPos());
+
     event.Skip();
```
Comment (vadz) [inline samples/splitter/splitter.cpp:565]: I'm surprised that calling `Skip()` still works, but if it does, it's actually a bug in `wxSplitterWindow`: if the event was skipped, it should _not_ be taken into account, i.e. there are 2 problems:

1. In `wxSplitterWindow` itself: it should ignore the position set by this event handler if it skips it.
2. In the sample: it shouldn't skip the event when handling it.

**Review by vadz — COMMENTED** [review]
Sorry, I think there is a relatively serious problem here, in addition to the other 2 minor remarks, could you please look at why how `Skip()` is handled? TIA!

File: samples/splitter/splitter.cpp (hunk 3 above)
Comment (skeetor) [reply samples/splitter/splitter.cpp:565]: The problem with Skip() is indeed a bug in `wxSplitterWindow`. The bug is with the `DoSendEvent` function. If the event is not vetoed, it is still allowed. So even if the handler was not found, it returns true because the event was not vetoed. I verified this in the sample with the `OnPositionChanging` handler which also uses Skip() and shows the same behavior. It was never noticed because the handler just prints the values, but doesn't change them.

When I added
```
event.SetSashPosition(100);
```
to it, it shows the same behavior as with resize, so this is an existing bug. I fix it now.

File: samples/splitter/splitter.cpp (hunk 2 above)
Comment (skeetor) [reply samples/splitter/splitter.cpp:280]: Fixed with the next commit.

File: samples/splitter/splitter.cpp (hunk 1 above)
Comment (skeetor) [reply samples/splitter/splitter.cpp:132]: Fixed with the next commit.

**Comment by vadz** [comment]
@skeetor You haven't pushed the next commit yet, have you?

File: samples/splitter/splitter.cpp (hunk 3 above)
Comment (skeetor) [reply samples/splitter/splitter.cpp]: Fixed bug with handling event.Skip() state and updated the samplecode to no longer call event.Skip().

**Comment by skeetor** [comment]
Just pushed it. I tested the bevavior also in the Changing event handler. If you set some hardcoded value you can see the difference. If you now call Skip() when changing it, the change is ignored, as it should be.

File: samples/splitter/splitter.cpp [hunk 4]
```diff
@@ -560,17 +558,19 @@ void MySplitterWindow::OnPositionResize(wxSplitterEvent &event)
...
-    event.Skip();
+    }
 }
 
 void MySplitterWindow::OnPositionChanging(wxSplitterEvent& event)
 {
     wxLogStatus(m_frame, "Position is changing, now = %d (or %d)",
                 event.GetSashPosition(), GetSashPosition());
-
-    event.Skip();
```
Comment (vadz) [inline samples/splitter/splitter.cpp:532]: Sorry, why was this removed? It should stay, we're not really processing the event, just logging it, so skipping it makes perfect sense.

**Review by vadz — COMMENTED** [review]
Thanks, almost perfect, but I think the removed `event.Skip()` should be restored -- please let me know if I'm missing anything.

File: samples/splitter/splitter.cpp (hunk 4 above)
Comment (vadz) [reply samples/splitter/splitter.cpp:532]: I'm going to readd this and merge if you don't have any objections.

Comment (skeetor) [reply samples/splitter/splitter.cpp:532]: No objection. Since the value is not changed, it doesn't matter if it is skipped or not. Maybe you should add a comment to that as well.

**Comment by vadz** [comment]
Sorry, I think there is another problem here: you've changed `DoSendEvent()` to not check for `event.IsAllowed()` any longer and inverted the meaning of its return value, but not all callers of it have been updated, notably `OnDoubleClickSash()`, i.e. `wxEVT_SPLITTER_DOUBLECLICKED` can't be vetoed any longer after your changes.

I've tried to fix this blindly and also removed `DoSendEvent()` entirely as it's useless now, please test the latest version. TIA!

**Comment by skeetor** [comment]
It works. I added a menu item so you can switch double clicking on or off now in the sample.

**Comment by vadz** [comment]
Thanks for testing, will merge soon.
